video/out/wayland_common.c

```diff
@@ -2098,6 +2131,26 @@ static int handle_round(int scale, int n)
     return (scale * n + WAYLAND_SCALE_FACTOR / 2) / WAYLAND_SCALE_FACTOR;
 }
 
+#if HAVE_FROG_PROTOCOLS
+static enum pl_color_primaries pl_primaries_to_frog(enum pl_color_primaries primaries)
+{
+    for (int i = 0; primaries_map[i].pl_primaries; i++) {
```

Comment:
To simplify this lookup, we could have direct lookup. Not biggie, but I even the overhead of full lut is not that big. As you prefer, either is good, just saying.

``` c
int transfer_map[PL_COLOR_PRIM_COUNT] = {
    [PL_COLOR_TRC_SRGB] = FROG_COLOR_MANAGED_SURFACE_TRANSFER_FUNCTION_SRGB,
    [PL_COLOR_TRC_GAMMA22] = FROG_COLOR_MANAGED_SURFACE_TRANSFER_FUNCTION_GAMMA_22,
    [PL_COLOR_TRC_PQ] = FROG_COLOR_MANAGED_SURFACE_TRANSFER_FUNCTION_ST2084_PQ,
};

static int pl_primaries_to_frog(enum pl_color_primaries primaries)
{
    return transfer_map[primaries];
}
```

Reply:
Ah I didn't realize we had a handy count enum. Yeah this is way better.